File: bench/benchmarks/sieve/sieve.py
N = 1000000
is_prime = bytearray(b'\x01') * (N + 1)
is_prime[0] = 0
is_prime[1] = 0

i = 2
while i * i <= N:
    if is_prime[i]:
        j = i * i
        while j <= N:
            is_prime[j] = 0
            j += i
    i += 1

count = 0
for i in range(2, N + 1):
    if is_prime[i]:
        count += 1
print(count)

File: bench/benchmarks/sieve/sieve.go
package main

import "fmt"

const N = 1000000

func main() {
	isPrime := make([]bool, N+1)
	for i := range isPrime {
		isPrime[i] = true
	}
	isPrime[0] = false
	isPrime[1] = false

	for i := 2; i*i <= N; i++ {
		if isPrime[i] {
			for j := i * i; j <= N; j += i {
				isPrime[j] = false
			}
		}
	}

	count := 0
	for i := 2; i <= N; i++ {
		if isPrime[i] {
			count++
		}
	}
	fmt.Println(count)
}
